runDistributable fails in v1.2.0-beta01

> ➜  ./gradlew runDistributable 
> Configuration on demand is an incubating feature.
> Type-safe project accessors is an incubating feature.
> > Task :desktopApp:createRuntimeImage FAILED
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> A problem was found with the configuration of task ':desktopApp:createRuntimeImage' (type 'AbstractJLinkTask').
>   - In plugin 'org.jetbrains.compose.ComposeCompilerKotlinSupportPlugin' type 'org.jetbrains.compose.desktop.application.tasks.AbstractJLinkTask' property 'javaRuntimePropertiesFile' specifies file '.../desktopApp/build/compose/tmp/checkRuntime/properties.bin' which doesn't exist.
>     
>     Reason: An input file was expected to be present but it doesn't exist.
>     
>     Possible solutions:
>       1. Make sure the file exists before the task is called.
>       2. Make sure that the task which produces the file is declared as an input.
>     
>     Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#input_file_does_not_exist for more details about this problem.
> 
> * Try:
> > Run with --stacktrace option to get the stack trace.
> > Run with --info or --debug option to get more log output.
> > Run with --scan to get full insights.
> 
> * Get more help at https://help.gradle.org
> 
> Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
> 
> You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
> 
> See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings
> 
> BUILD FAILED in 1s
> 9 actionable tasks: 1 executed, 3 from cache, 5 up-to-date
> 

Downgrading back to `v1.2.0-alpha01-dev774` works.

--- a/gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/configureJvmApplication.kt
+++ b/gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/configureJvmApplication.kt
@@ -58,7 +58,6 @@ private fun JvmApplicationContext.configureCommonJvmDesktopTasks(): CommonJvmDes
         taskNameObject = "runtime"
     ) {
         javaHome.set(app.javaHomeProvider)
-        javaRuntimePropertiesFile.set(jvmTmpDirForTask().file("properties.bin"))
     }
 
     val suggestRuntimeModules = tasks.register<AbstractSuggestModulesTask>(

--- a/gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/tasks/AbstractCheckNativeDistributionRuntime.kt
+++ b/gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/tasks/AbstractCheckNativeDistributionRuntime.kt
@@ -6,6 +6,7 @@
 package org.jetbrains.compose.desktop.application.tasks
 
 import org.gradle.api.file.Directory
+import org.gradle.api.file.RegularFile
 import org.gradle.api.file.RegularFileProperty
 import org.gradle.api.provider.Property
 import org.gradle.api.provider.Provider
@@ -23,14 +24,17 @@ internal const val MIN_JAVA_RUNTIME_VERSION = 15
 
 @CacheableTask
 abstract class AbstractCheckNativeDistributionRuntime : AbstractComposeDesktopTask() {
-    @get:Input
+    @get:PathSensitive(PathSensitivity.ABSOLUTE)
+    @get:InputDirectory
     val javaHome: Property<String> = objects.notNullProperty()
 
+    private val taskDir = project.layout.buildDirectory.dir("compose/tmp/$name")
+
     @get:OutputFile
-    val javaRuntimePropertiesFile: RegularFileProperty = objects.fileProperty()
+    val javaRuntimePropertiesFile: Provider<RegularFile> = taskDir.map { it.file("properties.bin") }
 
     @get:LocalState
-    val workingDir: Provider<Directory> = project.layout.buildDirectory.dir("compose/tmp/$name")
+    val workingDir: Provider<Directory> = taskDir.map { it.dir("localState") }
 
     private val javaExec: File
         get() = getTool("java")
@@ -47,6 +51,8 @@ abstract class AbstractCheckNativeDistributionRuntime : AbstractComposeDesktopTa
 
     @TaskAction
     fun run() {
+        taskDir.ioFile.mkdirs()
+
         val javaRuntimeVersion = try {
             getJavaRuntimeVersionUnsafe()?.toIntOrNull() ?: -1
         } catch (e: Exception) {

--- a/gradle-plugins/compose/src/test/kotlin/org/jetbrains/compose/test/tests/integration/DesktopApplicationTest.kt
+++ b/gradle-plugins/compose/src/test/kotlin/org/jetbrains/compose/test/tests/integration/DesktopApplicationTest.kt
@@ -123,6 +123,32 @@ class DesktopApplicationTest : GradlePluginTestBase() {
     }
 
     @Test
+    fun gradleBuildCache() = with(testProject(TestProjects.jvm)) {
+        modifyGradleProperties {
+            setProperty("org.gradle.caching", "true")
+        }
+        modifyTextFile("settings.gradle") {
+            it + "\n" + """
+                buildCache {
+                    local {
+                        directory = new File(rootDir, 'build-cache')
+                    }
+                }
+            """.trimIndent()
+        }
+
+        val packagingTask = ":packageDistributionForCurrentOS"
+        gradle(packagingTask).build().checks { check ->
+            check.taskOutcome(packagingTask, TaskOutcome.SUCCESS)
+        }
+
+        gradle("clean", packagingTask).build().checks { check ->
+            check.taskOutcome(":checkRuntime", TaskOutcome.FROM_CACHE)
+            check.taskOutcome(packagingTask, TaskOutcome.SUCCESS)
+        }
+    }
+
+    @Test
     fun packageMpp() = with(testProject(TestProjects.mpp)) {
         testPackageJvmDistributions()
     }

--- a/gradle-plugins/compose/src/test/kotlin/org/jetbrains/compose/test/utils/TestProject.kt
+++ b/gradle-plugins/compose/src/test/kotlin/org/jetbrains/compose/test/utils/TestProject.kt
@@ -8,6 +8,7 @@ package org.jetbrains.compose.test.utils
 import org.gradle.testkit.runner.GradleRunner
 import org.jetbrains.compose.desktop.application.internal.ComposeProperties
 import java.io.File
+import java.util.Properties
 
 data class TestEnvironment(
     val workingDir: File,
@@ -67,5 +68,29 @@ class TestProject(
 
     fun file(path: String): File =
         testEnvironment.workingDir.resolve(path)
+
+    fun modifyTextFile(path: String, fn: (String) -> String) {
+        val file = file(path)
+        val oldContent = file.readText()
+        val newContent = fn(oldContent)
+        file.writeText(newContent)
+    }
+    fun modifyGradleProperties(fn: Properties.() -> Unit) {
+        val propertiesFile = file("gradle.properties")
+        val properties = Properties()
+        if (propertiesFile.exists()) {
+            propertiesFile.bufferedReader().use { reader ->
+                properties.load(reader)
+            }
+        }
+        fn(properties)
+        propertiesFile.delete()
+
+        if (properties.isNotEmpty()) {
+            propertiesFile.bufferedWriter().use { writer ->
+                properties.store(writer, null)
+            }
+        }
+    }
 }
 

